lib/yubihsm_winhttp.c

```diff
@@ -313,17 +311,8 @@ static yh_rc backend_send_msg(yh_backend *connection, Msg *msg, Msg *response,
   }
 
   if (identifier != NULL && strlen(identifier) > 0 && strlen(identifier) < 32) {
-    tmp_identifier_len = mbstowcs(NULL, identifier, 0);
-    tmp_identifier = calloc(tmp_identifier_len + 1, sizeof(wchar_t));
-    if (tmp_identifier == NULL) {
-      DBG_ERR("Failed allocating memory for tmp_identifier");
-      free(context);
-      return YHR_MEMORY_ERROR;
-    }
-    mbstowcs(tmp_identifier, identifier, tmp_identifier_len + 1);
-    swprintf(hsm_identifier, 128, L"YubiHSM-Session: %ls", tmp_identifier);
+    swprintf(hsm_identifier, 64, L"YubiHSM-Session: %hs", identifier);
```

Comment:
What does the h do in the format string?

Reply:
It says that the argument is a non-wide char string.

Reply:
As a side-note %s is supposed to work as well according to ansi specifications, but Microsoft behaves differently, they take %s to mean wide char in wide-char functions while the ansi spec says that the formatting string should be interpreted the same in both cases. But %hs means char* in both environments. Microsoft also support %S which means the 'opposite' char type from the default.

Reply:
Not documented here https://en.cppreference.com/w/c/io/swprintf or here https://www.ibm.com/support/knowledgecenter/en/ssw_ibm_i_71/rtref/swprintf.htm

Ah, here it is: https://docs.microsoft.com/en-us/cpp/c-runtime-library/format-specification-syntax-printf-and-wprintf-functions?view=msvc-160